internal/transport/http2_server.go

```diff
@@ -1153,7 +1153,7 @@ func (t *http2Server) keepalive() {
 				if logger.V(logLevel) {
 					logger.Infof("transport: closing server transport due to maximum connection age.")
 				}
-				t.Close()
+				t.controlBuf.put(closeConnection{})
```

Comment:
`loopyWriter.closeConnectionHandler()` flushes the buffer and returns an error leading to `loopyWriter.run()` to return. But how and when does the actual transport get closed?

Would it be better to instead add a controlbuf operation to flush the buffer, and continue to call `t.Close()` here?

Reply:
We can't call `t.Close` here, because that won't wait for the controlBuf to exit, which is what we need.

This is how the transport will close now:

1. Loopy's run() will exit due to `closeConnection`, and get to here:

https://github.com/grpc/grpc-go/blob/110ed9e6ccbd391a691b24b6b918ce0991f7588f/internal/transport/http2_server.go#L339

2. That only closes the conn, not the transport.  The reader goroutine is responsible for closing the transport.  Closing the conn will lead to read errors in the reader goroutine, and that will lead to:

https://github.com/grpc/grpc-go/blob/110ed9e6ccbd391a691b24b6b918ce0991f7588f/internal/transport/http2_server.go#L632-L639

Reply:
This mirrors what we do in the client.  In the client, it's very important to read everything from the conn when it is closed (loopy may error before then), as it's possible the server sent response trailers for an RPC and immediately closed the connection; any RPCs that have messages/trailers that were written to the conn before closure must have those items read into their streams before the transport is closed.  On the server side, however, this seems less important to ensure, since the server would be unable to respond to any data on any remaining streams when the conn is closed.  Even though it's not as important, we probably should still do this, so the server application still has access to all the information we received, and it keeps things consistent between the two sides.

Reply:
Can we document this in `closeConnectionHandler()` or some place else?

Reply:
I think we had a race. My ask was for documenting how the transport eventually gets closed from `closeConnectionHandler`.

Reply:
Done!

Reply:
Thank you !! LGTM.